File: packages/flutter_shared/lib/tools/sort_tool.dart
/// 排序模式
enum SortMode { asc, desc, lengthAsc, lengthDesc, reverse }

/// 去重模式
enum DedupMode { all, caseInsensitive, trimThenDedup }

/// 文本排序 / 去重工具
class SortTool {
  SortTool._();

  /// 排序文本行
  static String sort(String input, SortMode mode) {
    final lines = input.split('\n');
    final trimmed = lines.map((l) => l.trimRight()).toList();

    switch (mode) {
      case SortMode.asc:
        trimmed.sort((a, b) => a.compareTo(b));
      case SortMode.desc:
        trimmed.sort((a, b) => b.compareTo(a));
      case SortMode.lengthAsc:
        trimmed.sort((a, b) => a.length.compareTo(b.length));
      case SortMode.lengthDesc:
        trimmed.sort((a, b) => b.length.compareTo(a.length));
      case SortMode.reverse:
        trimmed.sort((a, b) => 0);
        return trimmed.reversed.join('\n');
    }

    return trimmed.join('\n');
  }

  /// 去重文本行
  static String dedup(String input, DedupMode mode) {
    final lines = input.split('\n');
    final seen = <String>{};
    final result = <String>[];

    for (final line in lines) {
      final key = switch (mode) {
        DedupMode.all => line,
        DedupMode.caseInsensitive => line.toLowerCase(),
        DedupMode.trimThenDedup => line.trim(),
      };
      if (!seen.contains(key)) {
        seen.add(key);
        result.add(line);
      }
    }

    return result.join('\n');
  }

  /// 移除空行
  static String removeEmptyLines(String input) {
    return input.split('\n').where((l) => l.trim().isNotEmpty).join('\n');
  }

  /// 统计
  static ({int totalLines, int nonEmptyLines, int uniqueLines}) stats(String input) {
    final lines = input.split('\n');
    final total = lines.length;
    final nonEmpty = lines.where((l) => l.trim().isNotEmpty).length;
    final unique = lines.toSet().length;
    return (totalLines: total, nonEmptyLines: nonEmpty, uniqueLines: unique);
  }
}

File: miniapp/src/utils/tools/sortTool.ts
/**
 * 排序 / 去重工具
 * 移植自 flutter_shared/tools/sort_tool.dart
 */

export enum SortMode { asc = 'asc', desc = 'desc', lengthAsc = 'lengthAsc', lengthDesc = 'lengthDesc', reverse = 'reverse' }
export enum DedupMode { all = 'all', caseInsensitive = 'caseInsensitive', trimThenDedup = 'trimThenDedup' }

export interface SortStats {
  totalLines: number;
  nonEmptyLines: number;
  uniqueLines: number;
}

export const SortTool = {
  sort(input: string, mode: SortMode): string {
    const lines = input.split('\n');
    const trimmed = lines.map(l => l.trimEnd());

    switch (mode) {
      case SortMode.asc:
        trimmed.sort((a, b) => a.localeCompare(b));
        break;
      case SortMode.desc:
        trimmed.sort((a, b) => b.localeCompare(a));
        break;
      case SortMode.lengthAsc:
        trimmed.sort((a, b) => a.length - b.length);
        break;
      case SortMode.lengthDesc:
        trimmed.sort((a, b) => b.length - a.length);
        break;
      case SortMode.reverse:
        trimmed.reverse();
        break;
    }

    return trimmed.join('\n');
  },

  dedup(input: string, mode: DedupMode): string {
    const lines = input.split('\n');
    const seen = new Set<string>();
    const result: string[] = [];

    for (const line of lines) {
      let key: string;
      switch (mode) {
        case DedupMode.all:
          key = line;
          break;
        case DedupMode.caseInsensitive:
          key = line.toLowerCase();
          break;
        case DedupMode.trimThenDedup:
          key = line.trim();
          break;
      }
      if (!seen.has(key)) {
        seen.add(key);
        result.push(line);
      }
    }

    return result.join('\n');
  },

  removeEmptyLines(input: string): string {
    return input.split('\n').filter(l => l.trim().length > 0).join('\n');
  },

  stats(input: string): SortStats {
    const lines = input.split('\n');
    return {
      totalLines: lines.length,
      nonEmptyLines: lines.filter(l => l.trim().length > 0).length,
      uniqueLines: new Set(lines).size,
    };
  },
};
